api/src/main/java/io/strimzi/api/kafka/model/rebalance/KafkaRebalanceStatus.java

```diff
@@ -42,6 +43,16 @@ public void setOptimizationResult(Map<String, Object> optimizationResult) {
         this.optimizationResult = optimizationResult;
     }
 
+    @Description("A JSON object describing the progress of the rebalance")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public Map<String, Object> getProgress() {
+        return progress;
+    }
+
+    public void setProgress(Map<String, Object> progress) {
+        this.progress = progress;
+    }
```

Comment:
IIRC the proposal expects here only a link to the ConfigMap. So this is likely a wrong type and description?

Reply:
Argh you are correct, from the proposal [1]:

> This proposal extends the status section of the `KafkaRebalance` custom resource to include a progress section with a nested `rebalanceProgressConfigMap` field to reference that same `ConfigMap`.

I was hoping take some liberties with the description here and use a map to hold the nested  `rebalanceProgressConfigMap` field. I figured this would give us more more flexibility with the API in the future.

WDYT? If I updated the description to something like "A Map with fields describing the progress of the rebalance" in the code, do you think that would be ok? Or would you prefer I remove the map in favor of a single `rebalanceProgressConfigMap` field?

[1] https://github.com/strimzi/proposals/blob/main/098-rebalance-progress-status.md

Reply:
Sorry, but I don't understand what flexibility you are talking about here. Flexibility is not about what Java type you use, but about how the API looks like. So as long as we still want to use this:

```yaml
status:
  # ...
  progress: [1]
    rebalanceProgressConfigMap: my-rebalance [2]
```

It should be modelled that way in the API to give us proper documentation and validation. If it needs to be changed one day, it is not about what type it has in the Java code but about what the API structure is compared to what we promised.

If the YAML above does not represent anymore what you want to do, then it should be discussed on a higher level - for example in the proposal level or on Slack. And not done silently in the implementation PR.

Reply:
Thanks Jakub, you are absolutely right.  Just updated the type and descriptions to match what is in the proposal.